### ISSUE TITLE
load assetbundle error

### ISSUE THREAD

**Issue by dnaroma** [HearthSim/UnityPack#35]
I'm trying loading a assetbundle with header `UnityFS 5.x.x 5.4.1f1` but got error

```
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    print(id, object.type)
  File "/home/tangpei/文档/UnityPack/unitypack/object.py", line 28, in type
    script = self.read()["m_Script"]
  File "/home/tangpei/文档/UnityPack/unitypack/object.py", line 74, in read
    return self.read_value(self.type_tree, buf)
  File "/home/tangpei/文档/UnityPack/unitypack/object.py", line 132, in read_value
    result[child.name] = self.read_value(child, buf)
  File "/home/tangpei/文档/UnityPack/unitypack/object.py", line 101, in read_value
    result = buf.read_string(size)
  File "/home/tangpei/文档/UnityPack/unitypack/utils.py", line 70, in read_string
    ret = struct.unpack(self.endian + "%is" % (size), self.read(size))[0]
struct.error: unpack requires a bytes object of length 3410181934
```

[001_school_winter.zip](https://github.com/HearthSim/UnityPack/files/858834/001_school_winter.zip)

**Comment by Mactastic1-5** [comment]
If you read the last line, it says the object has to be the bytes length of 3.4GB. I didn't look at the file.

**Comment by dnaroma** [comment]
@Stellarspace Thanks. I think it may got wrong size from file... I tried other files, they didn't crash but the size is also huge.

**Comment by Mactastic1-5** [comment]
@tangpei506 It's weird that developers compile their Unity games and apps with a large file size. Is it a desktop game?

**Comment by robert-nix** [comment]
This is just a decode issue.  The asset itself should be 4.5MB uncompressed, and the assetbundle uses LZ4 compression.  The 3410181934 number is a uint32 (2E 3F 43 CB) visible in the compressed bytes.  Unsure why unitypack isn't feeding it through LZ4 properly.

**Comment by robert-nix** [comment]
Can you pull master and see if this is fixed now?

**Comment by robert-nix** [comment]
This was fixed by 11c22bcc.

**Comment by dnaroma** [comment]
@Mischanix Thanks!
